[ALLUXIO-1942 ] Better support parquet file in Alluxio

peisun1115 opened the pull request:
Better support parquet file in Alluxio: 
a) handle cache stream related exceptions properly in FileInStream 
b) Increase netty network timeout from 3s to 30s
c) Fix a bug related to root dir in FileSystemMaster

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9713/
Test PASSed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9714/
Test PASSed.

AmplabJenkins commented:
Build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9722/<h2>Build result: FAILURE</span></h2>[...truncated 1311 lines...][JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/s3/target/alluxio-underfs-s3-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-s3/1.1.0-SNAPSHOT/alluxio-underfs-s3-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/s3/target/alluxio-underfs-s3-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-s3/1.1.0-SNAPSHOT/alluxio-underfs-s3-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/assembly/pom.xml to org.alluxio/alluxio-assemblies/1.1.0-SNAPSHOT/alluxio-assemblies-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/shell/pom.xml to org.alluxio/alluxio-shell/1.1.0-SNAPSHOT/alluxio-shell-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/oss/pom.xml to org.alluxio/alluxio-underfs-oss/1.1.0-SNAPSHOT/alluxio-underfs-oss-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/oss/target/alluxio-underfs-oss-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-oss/1.1.0-SNAPSHOT/alluxio-underfs-oss-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/oss/target/alluxio-underfs-oss-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-oss/1.1.0-SNAPSHOT/alluxio-underfs-oss-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/client/pom.xml to org.alluxio/alluxio-keyvalue-client/1.1.0-SNAPSHOT/alluxio-keyvalue-client-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/hdfs/pom.xml to org.alluxio/alluxio-underfs-hdfs/1.1.0-SNAPSHOT/alluxio-underfs-hdfs-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/hdfs/target/alluxio-underfs-hdfs-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-hdfs/1.1.0-SNAPSHOT/alluxio-underfs-hdfs-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/hdfs/target/alluxio-underfs-hdfs-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-hdfs/1.1.0-SNAPSHOT/alluxio-underfs-hdfs-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/swift/pom.xml to org.alluxio/alluxio-underfs-swift/1.1.0-SNAPSHOT/alluxio-underfs-swift-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/swift/target/alluxio-underfs-swift-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-swift/1.1.0-SNAPSHOT/alluxio-underfs-swift-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/swift/target/alluxio-underfs-swift-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-swift/1.1.0-SNAPSHOT/alluxio-underfs-swift-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/examples/pom.xml to org.alluxio/alluxio-examples/1.1.0-SNAPSHOT/alluxio-examples-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/core/server/pom.xml to org.alluxio/alluxio-core-server/1.1.0-SNAPSHOT/alluxio-core-server-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/common/pom.xml to org.alluxio/alluxio-keyvalue-common/1.1.0-SNAPSHOT/alluxio-keyvalue-common-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/pom.xml to org.alluxio/alluxio-underfs/1.1.0-SNAPSHOT/alluxio-underfs-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/glusterfs/pom.xml to org.alluxio/alluxio-underfs-glusterfs/1.1.0-SNAPSHOT/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/glusterfs/target/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-glusterfs/1.1.0-SNAPSHOT/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/glusterfs/target/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-glusterfs/1.1.0-SNAPSHOT/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/integration/pom.xml to org.alluxio/alluxio-integration/1.1.0-SNAPSHOT/alluxio-integration-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/pom.xml to org.alluxio/alluxio-keyvalue/1.1.0-SNAPSHOT/alluxio-keyvalue-1.1.0-SNAPSHOT.pomchannel stopped
Test FAILed.

AmplabJenkins commented:
Build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9725/
Test PASSed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9781/
Test PASSed.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9801/<h2>Build result: FAILURE</span></h2>[...truncated 1336 lines...][JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/s3/target/alluxio-underfs-s3-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-s3/1.1.0-SNAPSHOT/alluxio-underfs-s3-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/s3/target/alluxio-underfs-s3-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-s3/1.1.0-SNAPSHOT/alluxio-underfs-s3-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/assembly/pom.xml to org.alluxio/alluxio-assemblies/1.1.0-SNAPSHOT/alluxio-assemblies-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/shell/pom.xml to org.alluxio/alluxio-shell/1.1.0-SNAPSHOT/alluxio-shell-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/oss/pom.xml to org.alluxio/alluxio-underfs-oss/1.1.0-SNAPSHOT/alluxio-underfs-oss-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/oss/target/alluxio-underfs-oss-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-oss/1.1.0-SNAPSHOT/alluxio-underfs-oss-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/oss/target/alluxio-underfs-oss-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-oss/1.1.0-SNAPSHOT/alluxio-underfs-oss-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/client/pom.xml to org.alluxio/alluxio-keyvalue-client/1.1.0-SNAPSHOT/alluxio-keyvalue-client-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/hdfs/pom.xml to org.alluxio/alluxio-underfs-hdfs/1.1.0-SNAPSHOT/alluxio-underfs-hdfs-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/hdfs/target/alluxio-underfs-hdfs-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-hdfs/1.1.0-SNAPSHOT/alluxio-underfs-hdfs-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/hdfs/target/alluxio-underfs-hdfs-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-hdfs/1.1.0-SNAPSHOT/alluxio-underfs-hdfs-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/swift/pom.xml to org.alluxio/alluxio-underfs-swift/1.1.0-SNAPSHOT/alluxio-underfs-swift-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/swift/target/alluxio-underfs-swift-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-swift/1.1.0-SNAPSHOT/alluxio-underfs-swift-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/swift/target/alluxio-underfs-swift-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-swift/1.1.0-SNAPSHOT/alluxio-underfs-swift-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/examples/pom.xml to org.alluxio/alluxio-examples/1.1.0-SNAPSHOT/alluxio-examples-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/core/server/pom.xml to org.alluxio/alluxio-core-server/1.1.0-SNAPSHOT/alluxio-core-server-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/common/pom.xml to org.alluxio/alluxio-keyvalue-common/1.1.0-SNAPSHOT/alluxio-keyvalue-common-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/pom.xml to org.alluxio/alluxio-underfs/1.1.0-SNAPSHOT/alluxio-underfs-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/glusterfs/pom.xml to org.alluxio/alluxio-underfs-glusterfs/1.1.0-SNAPSHOT/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/glusterfs/target/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-glusterfs/1.1.0-SNAPSHOT/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/glusterfs/target/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-glusterfs/1.1.0-SNAPSHOT/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/integration/pom.xml to org.alluxio/alluxio-integration/1.1.0-SNAPSHOT/alluxio-integration-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/pom.xml to org.alluxio/alluxio-keyvalue/1.1.0-SNAPSHOT/alluxio-keyvalue-1.1.0-SNAPSHOT.pomchannel stopped
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9803/
Test PASSed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9806/
Test PASSed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9807/
Test PASSed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9808/
Test PASSed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9809/
Test PASSed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9817/
Test PASSed.

peisun1115 commented on core/server/src/main/java/alluxio/master/file/FileSystemMaster.java:
```diff
@@ -1630,7 +1630,7 @@ private long loadDirectoryMetadata(AlluxioURI path, LoadMetadataOptions options)
     if (mInodeTree.inodePathExists(path)) {
       Inode inode = mInodeTree.getInodeByPath(path);
-      if (inode.isPersisted()) {
+      if (inode.isPersisted() || path.isRoot()) {
```
path.isRoot() is necessary because Alluxio considers root as existing by default.

apc999 replied:
please let @gpang  verify  this change

peisun1115 replied:
I will send another PR for this.

peisun1115 commented:
@calvinjia  @apc999 

I have rebased the PR to master. Now it is ready for review.

peisun1115 commented on core/common/src/main/resources/alluxio-default.properties:
```diff
@@ -148,14 +148,14 @@ alluxio.user.failed.space.request.limits=3
 alluxio.user.file.master.client.threads=10
 alluxio.user.file.readtype.default=CACHE_PROMOTE
-alluxio.user.file.cache.partially.read.block=false
+alluxio.user.file.cache.partially.read.block=true
```
I have changed this to true by default because I think it is annoying to ask spark users to set this to true everytime. Let me know what you think. thanks

apc999 commented on core/client/src/main/java/alluxio/client/block/LocalBlockOutStream.java:
```diff
@@ -136,4 +138,12 @@ private long requestSpace(long requestBytes) throws IOException {
   }
+
+  /**
+   * Release {@link #mBlockWorkerClient} and set {@link #mClosed} to true.
```
NIT: Releases

peisun1115 replied:
done

apc999 commented on core/client/src/main/java/alluxio/client/file/FileInStream.java:
```diff
@@ -124,6 +126,9 @@ protected FileInStream(URIStatus status, InStreamOptions options) {
     mShouldCache = mAlluxioStorageType.isStore();
     mShouldCachePartiallyReadBlock = options.isCachePartiallyReadBlock();
+    if (mShouldCachePartiallyReadBlock) {
+      LOG.debug("Should cache partially read block enabled.");
```
NIT: `String.format("%s enabled", Constants.USER_FILE_CACHE_PARTIALLY_READ_BLOCK)`

peisun1115 replied:
done

apc999 commented on core/client/src/main/java/alluxio/client/file/FileInStream.java:
```diff
@@ -321,17 +326,25 @@ protected boolean shouldUpdateStreams(long currentBlockId) {
+      } else if (e.getCause() instanceof InvalidWorkerStateException) {
+        LOG.info("Block {} has invalid worker state when being cancelled.", getCurrentBlockId());
+      } else {
+        LOG.info("Close or cancel throws IOExecption {}.", e.getMessage());
```
shall we escalate the exception there?

peisun1115 replied:
I changed this to LOG.warn now.

I don't think we should throw an exception here because read should continue though cache fails (e.g. timeout because of slow network, server unreachable because of alluxio worker is busy etc).  What do you think?

apc999 replied:
My concern is that, I am not sure if the caller of `closeOrCancelCacheStream` making any assumption here: is it possible that previously the caller assume exceptions thrown to indicate the error, while now this assumption is no more true? Please double-check if callers of `closeOrCancelCacheStream` did make this assumption.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9818/<h2>Build result: FAILURE</span></h2>[...truncated 1412 lines...][JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/s3/target/alluxio-underfs-s3-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-s3/1.1.0-SNAPSHOT/alluxio-underfs-s3-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/s3/target/alluxio-underfs-s3-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-s3/1.1.0-SNAPSHOT/alluxio-underfs-s3-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/assembly/pom.xml to org.alluxio/alluxio-assemblies/1.1.0-SNAPSHOT/alluxio-assemblies-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/shell/pom.xml to org.alluxio/alluxio-shell/1.1.0-SNAPSHOT/alluxio-shell-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/oss/pom.xml to org.alluxio/alluxio-underfs-oss/1.1.0-SNAPSHOT/alluxio-underfs-oss-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/oss/target/alluxio-underfs-oss-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-oss/1.1.0-SNAPSHOT/alluxio-underfs-oss-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/oss/target/alluxio-underfs-oss-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-oss/1.1.0-SNAPSHOT/alluxio-underfs-oss-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/client/pom.xml to org.alluxio/alluxio-keyvalue-client/1.1.0-SNAPSHOT/alluxio-keyvalue-client-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/hdfs/pom.xml to org.alluxio/alluxio-underfs-hdfs/1.1.0-SNAPSHOT/alluxio-underfs-hdfs-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/hdfs/target/alluxio-underfs-hdfs-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-hdfs/1.1.0-SNAPSHOT/alluxio-underfs-hdfs-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/hdfs/target/alluxio-underfs-hdfs-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-hdfs/1.1.0-SNAPSHOT/alluxio-underfs-hdfs-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/swift/pom.xml to org.alluxio/alluxio-underfs-swift/1.1.0-SNAPSHOT/alluxio-underfs-swift-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/swift/target/alluxio-underfs-swift-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-swift/1.1.0-SNAPSHOT/alluxio-underfs-swift-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/swift/target/alluxio-underfs-swift-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-swift/1.1.0-SNAPSHOT/alluxio-underfs-swift-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/examples/pom.xml to org.alluxio/alluxio-examples/1.1.0-SNAPSHOT/alluxio-examples-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/core/server/pom.xml to org.alluxio/alluxio-core-server/1.1.0-SNAPSHOT/alluxio-core-server-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/common/pom.xml to org.alluxio/alluxio-keyvalue-common/1.1.0-SNAPSHOT/alluxio-keyvalue-common-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/pom.xml to org.alluxio/alluxio-underfs/1.1.0-SNAPSHOT/alluxio-underfs-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/glusterfs/pom.xml to org.alluxio/alluxio-underfs-glusterfs/1.1.0-SNAPSHOT/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/glusterfs/target/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-glusterfs/1.1.0-SNAPSHOT/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/glusterfs/target/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-glusterfs/1.1.0-SNAPSHOT/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/integration/pom.xml to org.alluxio/alluxio-integration/1.1.0-SNAPSHOT/alluxio-integration-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/pom.xml to org.alluxio/alluxio-keyvalue/1.1.0-SNAPSHOT/alluxio-keyvalue-1.1.0-SNAPSHOT.pomchannel stopped
Test FAILed.

peisun1115 commented:
Jenkins, test this please

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9821/<h2>Build result: ABORTED</span></h2>[...truncated 1293 lines...]Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.004 secRunning alluxio.client.file.policy.SpecificHostPolicyTestTests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 secRunning alluxio.client.file.policy.RoundRobinPolicyTestTests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 secRunning alluxio.client.file.policy.LocalFirstPolicyTestTests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.013 secRunning alluxio.client.file.policy.MostAvailableFirstPolicyTestTests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 secRunning alluxio.client.file.URIStatusTestTests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.016 secRunning alluxio.client.file.FileSystemContextTestTests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.517 secRunning alluxio.client.file.FileInStreamTestBuild timed out (after 60 minutes). Marking the build as aborted.Results :Tests run: 19, Failures: 0, Errors: 0, Skipped: 0[ERROR] There was a timeout or other error in the forkBuild was abortedchannel stopped[WARNING] Failed to notify spy hudson.maven.Maven3Builder$JenkinsEventSpy: Invalid object ID 13 iota=35
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9823/
Test PASSed.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9822/<h2>Build result: ABORTED</span></h2>[...truncated 1328 lines...]	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)	at java.lang.reflect.Method.invoke(Method.java:606)	at org.codehaus.plexus.classworlds.launcher.Launcher.launchStandard(Launcher.java:330)	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:238)	at jenkins.maven3.agent.Maven31Main.launch(Maven31Main.java:181)	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)	at java.lang.reflect.Method.invoke(Method.java:606)	at hudson.maven.Maven3Builder.call(Maven3Builder.java:136)	at hudson.maven.Maven3Builder.call(Maven3Builder.java:71)	at hudson.remoting.UserRequest.perform(UserRequest.java:120)	at hudson.remoting.UserRequest.perform(UserRequest.java:48)	at hudson.remoting.Request$2.run(Request.java:326)	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)	at java.util.concurrent.FutureTask.run(FutureTask.java:262)	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)	at java.lang.Thread.run(Thread.java:745)[INFO] [INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ alluxio-core-client ---channel stopped
Test FAILed.

peisun1115 commented:
Jenkins, test this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9826/
Test PASSed.

apc999 commented on core/client/src/main/java/alluxio/client/block/LocalBlockOutStream.java:
```diff
@@ -81,9 +81,9 @@ public void cancel() throws IOException {
       mBlockWorkerClient.cancelBlock(mBlockId);
     } catch (AlluxioException e) {
       throw new IOException(e);
+    } finally {
```
Q: was this previously a bug?

peisun1115 replied:
I think so.

apc999 replied:
ack

apc999 commented on core/client/src/main/java/alluxio/client/block/LocalBlockOutStream.java:
```diff
@@ -98,11 +98,13 @@ public void close() throws IOException {
         mBlockWorkerClient.cacheBlock(mBlockId);
       } catch (AlluxioException e) {
         throw new IOException(e);
+      } finally {
```
Q: was this previously a bug?

peisun1115 replied:
i think so

apc999 replied:
ack

apc999 commented on core/client/src/main/java/alluxio/client/file/FileInStream.java:
```diff
@@ -249,7 +254,7 @@ public long skip(long n) throws IOException {
 
     long toSkip = Math.min(n, remaining());
-    seekInternal(mPos + toSkip);
+    seek(mPos + toSkip);
```
why do we change from `seekInternal` to `seek`?

peisun1115 replied:
Because I think the partially read block caching should be implemented in skip as well.  Previously skip doesn't have this feature. I have not seen a problem related to skip but maybe spark use skip in the future. that will make it very hard to debug if we don't have this

apc999 replied:
ack

apc999 commented on core/client/src/main/java/alluxio/client/file/FileInStream.java:
```diff
@@ -321,17 +326,28 @@ protected boolean shouldUpdateStreams(long currentBlockId) {
+      }
+    } catch (IOException e) {
+      if (e.getCause() instanceof BlockDoesNotExistException) {
+        LOG.info("Block {} does not exist when being cancelled.", getCurrentBlockId());
```
i think the three cases should be `LOG.warn`, while the last case shoule be `LOG.error`

apc999 replied:
and could you also comment here why for these three cases, it is fine to proceed. E.g., what could actually be happening to cause these problem?

apc999 replied:
Another dumb question, I guess the change here is optimized for concurrent Alluxio readers (with cache turned on), e.g., Parquet readers. Will this change affect the normal case?

peisun1115 replied:
1. If I use LOG.warn or LOG.error, the spark shell will be spammed by logs unless the user do sc.setLogLevel("ERROR").  In fact, i think this should be LOG.debug. Otherwise it might affect performance since this it happens so often in large parquet files.  
2. Documents added. 
3. This change optimizes for concurrent Alluxio readers with cache turned on (note that it affects concurrent readers with or without partially read block caching turned on). It affects normal cases. We used to throw and exception to the reader if the caching fails (can be many reasons such as network partition, unreachable due to alluxio worker is busy etc. etc. ), but now we don't.   I think it is expected to see alluxio client to behave this way, i.e. don't crash the reader if cache fails.  For example, a spark user might prefer to have his job working instead of crashing if one AlluxioWorker lost.

peisun1115 replied:
Another point I forgot to mention:
Before this PR, we were inconsistent in handling cache stream related exceptions. We were ignoring all exceptions (only log them) in cache.write but not in cache.cancel or cache.close. I think it is better to be consistent.

apc999 replied:
@peisun1115 , about the expected behavior on alluxio client, please double check on Calvin as he has most context of alluxio client-side code.

apc999 commented:
I left a bunch questions. Thanks. Now I agree with Calvin that it might be worth making a subclass of FileInStream that can cache partially-read blocks. I don't think it should be done in this PR, but I will view this as a tech debt.

peisun1115 commented:
@apc999  I have addressed your comments.

About subclassing FileInStream, I talked with Calvin last time. Personally I think we are abusing Inheritance because:
1.  CachePartiallyReadBlock is only one option in FileInStream. When we have multiple options that you want to subclass, that becomes problematic since we might want to have a combination of the options.  For example, now we want to have UnknownLengthFileInStream with partially read block caching enabled. Do you want to have a subclass of FileInStream and a subclass of UnknownLengthFileInStream?
2.  Quoting from effective java (Item 16: Unlike method invocation, inheritance violates encapsulation) 
3. When I make changes to FileInStream, I felt that the subclass UnknownLengthFileInStream makes it harder to make changes to FileInStream if UnknownLengthFileInStream overrides too many small functions. 

I think  this is a style of programming, we can discuss more and I can be convinced if you and clears my concerns.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9830/
Test PASSed.

apc999 commented on core/client/src/main/java/alluxio/client/file/FileInStream.java:
```diff
@@ -321,17 +326,40 @@ protected boolean shouldUpdateStreams(long currentBlockId) {
+      }
+    } catch (IOException e) {
+      if (e.getCause() instanceof BlockDoesNotExistException) {
+        // This happens if two concurrent readers read trying to cache the same block. One cancelled
```
thanks for the detailed documentation here! it is very helpful

calvinjia replied:
For the log messages, could you make this a bit more user facing? ie. describe the expected state (if the block is expected to be cached or not)

apc999 commented:
thanks for the update @peisun1115 . I have no more comments left, but leaving comments about consulting with @calvinjia and @gpang (it seems you have separate PR for him). I will watch the conversation then.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9838/
Test PASSed.

calvinjia commented on core/client/src/main/java/alluxio/client/file/FileInStream.java:
```diff
@@ -124,6 +126,9 @@ protected FileInStream(URIStatus status, InStreamOptions options) {
     mShouldCachePartiallyReadBlock = options.isCachePartiallyReadBlock();
+    if (mShouldCachePartiallyReadBlock) {
+      LOG.debug("CacheParitallyReadBlock is enabled.");
+    }
```
If we have a debug message about the stream options, perhaps use the `options.toString` method to get all the info?

peisun1115 replied:
Done

calvinjia commented on core/client/src/main/java/alluxio/client/file/FileInStream.java:
```diff
@@ -353,11 +381,14 @@ private long getCurrentBlockId() {
    * Logs IO exceptions thrown in response to the worker cache request. If the exception is not an
    * expected exception, a warning will be logged with the stack trace.
```
Update javadoc

peisun1115 replied:
done

calvinjia commented on core/client/src/main/java/alluxio/client/file/FileInStream.java:
```diff
@@ -353,11 +381,14 @@ private long getCurrentBlockId() {
+      LOG.info(BLOCK_ID_EXISTS_SO_NOT_CACHED, getCurrentBlockId());
     } else {
-      LOG.warn(BLOCK_ID_NOT_CACHED, getCurrentBlockId(), e);
+      LOG.info(BLOCK_ID_NOT_CACHED, getCurrentBlockId());
```
Why is this changed to `info`?

peisun1115 replied:
It prints some unnecessary msgs on the spark-shell. I changed this back to LOG.warn but removed the exception in it.  I forgot the exact situation this happened. I might need to change this back to info after the test. For now, it is warn

calvinjia commented on core/client/src/main/java/alluxio/client/file/FileInStream.java:
```diff
@@ -580,9 +611,12 @@ private void readCurrentBlockToPos(long pos) throws IOException {
 
-    byte[] buffer = new byte[Math.min(Constants.MB, (int) len)];
+    // Do not set the buffer size too small to avoid slowing down seek by too much.
+    byte[] buffer = new byte[Math.min(Constants.MB * 128, (int) len)];
```
Make this value configurable?

peisun1115 replied:
done

peisun1115 commented:
@calvinjia  I have addressed your comments. PTAL. Thank you

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9872/<h2>Build result: FAILURE</span></h2>[...truncated 1312 lines...][JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/s3/target/alluxio-underfs-s3-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-s3/1.1.0-SNAPSHOT/alluxio-underfs-s3-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/s3/target/alluxio-underfs-s3-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-s3/1.1.0-SNAPSHOT/alluxio-underfs-s3-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/assembly/pom.xml to org.alluxio/alluxio-assemblies/1.1.0-SNAPSHOT/alluxio-assemblies-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/shell/pom.xml to org.alluxio/alluxio-shell/1.1.0-SNAPSHOT/alluxio-shell-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/oss/pom.xml to org.alluxio/alluxio-underfs-oss/1.1.0-SNAPSHOT/alluxio-underfs-oss-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/oss/target/alluxio-underfs-oss-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-oss/1.1.0-SNAPSHOT/alluxio-underfs-oss-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/oss/target/alluxio-underfs-oss-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-oss/1.1.0-SNAPSHOT/alluxio-underfs-oss-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/client/pom.xml to org.alluxio/alluxio-keyvalue-client/1.1.0-SNAPSHOT/alluxio-keyvalue-client-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/hdfs/pom.xml to org.alluxio/alluxio-underfs-hdfs/1.1.0-SNAPSHOT/alluxio-underfs-hdfs-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/hdfs/target/alluxio-underfs-hdfs-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-hdfs/1.1.0-SNAPSHOT/alluxio-underfs-hdfs-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/hdfs/target/alluxio-underfs-hdfs-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-hdfs/1.1.0-SNAPSHOT/alluxio-underfs-hdfs-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/swift/pom.xml to org.alluxio/alluxio-underfs-swift/1.1.0-SNAPSHOT/alluxio-underfs-swift-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/swift/target/alluxio-underfs-swift-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-swift/1.1.0-SNAPSHOT/alluxio-underfs-swift-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/swift/target/alluxio-underfs-swift-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-swift/1.1.0-SNAPSHOT/alluxio-underfs-swift-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/examples/pom.xml to org.alluxio/alluxio-examples/1.1.0-SNAPSHOT/alluxio-examples-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/core/server/pom.xml to org.alluxio/alluxio-core-server/1.1.0-SNAPSHOT/alluxio-core-server-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/common/pom.xml to org.alluxio/alluxio-keyvalue-common/1.1.0-SNAPSHOT/alluxio-keyvalue-common-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/pom.xml to org.alluxio/alluxio-underfs/1.1.0-SNAPSHOT/alluxio-underfs-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/glusterfs/pom.xml to org.alluxio/alluxio-underfs-glusterfs/1.1.0-SNAPSHOT/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/glusterfs/target/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-glusterfs/1.1.0-SNAPSHOT/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/glusterfs/target/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-glusterfs/1.1.0-SNAPSHOT/alluxio-underfs-glusterfs-1.1.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/integration/pom.xml to org.alluxio/alluxio-integration/1.1.0-SNAPSHOT/alluxio-integration-1.1.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/pom.xml to org.alluxio/alluxio-keyvalue/1.1.0-SNAPSHOT/alluxio-keyvalue-1.1.0-SNAPSHOT.pomchannel stopped
Test FAILed.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9873/
Test PASSed.

calvinjia commented:
@peisun1115 Looks good overall, could you add the new option fields to `InStreamOptionsTest`?

peisun1115 commented:
@calvinjia  tests updated

calvinjia commented:
LGTM

apc999 commented on docs/_data/table/cn/user-configuration.yml:
```diff
@@ -27,6 +27,8 @@ alluxio.user.file.writetype.default:
 alluxio.user.file.cache.partially.read.block:
   当读取类型是 `CACHE` 或者 `CACHE_PROMOTE` 的时候, 如果设置这个变量为真, 没有完全读取的数据块也会被存到Alluxio内.
+alluxio.user.file.seek.buffer.size.bytes:
+  在文件seek操作中使用的缓存大小。
```
also indicate that this is only used when alluxio.user.file.cache.partially.read.block is enabled?

peisun1115 replied:
done

apc999 commented:
one minor format comment, otherwise, LGTM

peisun1115 commented:
@apc999  Addressed

apc999 commented:
LGTM

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/9882/
Test PASSed.
